src/MSBuild/XMake.cs

```diff
@@ -3210,8 +3210,12 @@ private static string ProcessToolsVersionSwitch(string[] parameters)
                 originalVerbosity = ProcessVerbositySwitch(verbositySwitchParameters[verbositySwitchParameters.Length - 1]);
                 verbosity = originalVerbosity;
             }
+            var loggers = new List<ILogger>();
+
+            var outVerbosity = verbosity;
+            ProcessBinaryLogger(binaryLoggerParameters, loggers, ref outVerbosity);
```

Comment:
I reordered loggers initialization to have binlog on top - to ensure it receives all the events from initialiation of custom loggers.

@rainersigwald - is there any hidden risk associated to reordering the creation and initialiation of the loggers

Reply:
Nothing springs to mind.